opentracing-java8/src/main/java/opentracing/propagation/AbstractTracer.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing.propagation;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import opentracing.Span;
+import opentracing.Tracer;
+import opentracing.propagation.Extractor;
+import opentracing.propagation.Injector;
+import opentracing.propagation.SpanImpl;
+
+abstract class AbstractTracer implements Tracer {
```

Comment:
yes, as described in the original description that is the purpose of this PR as opposed to PR#10. Although there are some things in PR#10 that can get moved to this PR.